File: Freedom/支付宝/我的/SDAssetsTableViewController.m
//  SDAssetsTableViewController.m
//  Freedom
//  Created by Super on 15-6-4.

#import "SDAssetsTableViewController.h"
#import "SDAssetsTableViewControllerCellModel.h"
#import "SDYuEBaoTableViewController.h"
#import "SDAssetsTableViewControllerCellModel.h"
#import "SDBasicTableViewController.h"
@interface SDAssetsTableViewControllerCell : SDBasicTableViewControllerCell
@end
@implementation SDAssetsTableViewControllerCell
- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        self.textLabel.textColor = [UIColor darkGrayColor];
        self.textLabel.font = [UIFont systemFontOfSize:15];
    }return self;
}
- (void)setModel:(NSObject *)model{
    [super setModel:model];
    SDAssetsTableViewControllerCellModel *cellModel = (SDAssetsTableViewControllerCellModel *)model;
    self.textLabel.text = cellModel.title;
    self.imageView.image = [UIImage imageNamed:cellModel.iconImageName];
    self.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
}
@end
@implementation SDAssetsTableViewController
- (void)viewDidLoad{
    [super viewDidLoad];
    self.navigationItem.title = @"我的";
    self.sectionsNumber = 3;
    self.cellClass = [SDAssetsTableViewControllerCell class];
    [self setupModel];
    
    UIView *header = [[UIView alloc] init];
    
    self.tableView.tableHeaderView = header;
    
}
- (void)setupModel{
    // section 0 的model
    SDAssetsTableViewControllerCellModel *model01 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"余额宝" iconImageName:@"20000032Icon" destinationControllerClass:[SDYuEBaoTableViewController class]];
    SDAssetsTableViewControllerCellModel *model02 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"招财宝" iconImageName:@"20000059Icon" destinationControllerClass:[SDBasicTableViewController class]];
    
    SDAssetsTableViewControllerCellModel *model03 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"娱乐宝" iconImageName:@"20000077Icon" destinationControllerClass:[SDBasicTableViewController class]];
    
    // section 1 的model
    SDAssetsTableViewControllerCellModel *model11 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"芝麻信用分" iconImageName:@"20000118Icon" destinationControllerClass:[SDBasicTableViewController class]];
    
    SDAssetsTableViewControllerCellModel *model12 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"随身贷" iconImageName:@"20000180Icon" destinationControllerClass:[SDBasicTableViewController class]];
    
    SDAssetsTableViewControllerCellModel *model13 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"我的保障" iconImageName:@"20000110Icon" destinationControllerClass:[SDBasicTableViewController class]];
    
    // section 2 的model
    SDAssetsTableViewControllerCellModel *model21 = [SDAssetsTableViewControllerCellModel modelWithTitle:@"爱心捐赠" iconImageName:@"09999978Icon" destinationControllerClass:[SDBasicTableViewController class]];
    self.dataArray = @[@[model01, model02, model03],
                       @[model11, model12, model13],
                       @[model21]
                       ];
}
#pragma mark - delegate 
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    SDAssetsTableViewControllerCellModel *model = [self.dataArray[indexPath.section] objectAtIndex:indexPath.row];
    UIViewController *vc = [[model.destinationControllerClass alloc] init];
    vc.title = model.title;
    [self.navigationController pushViewController:vc animated:YES];
}
- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section{
    return 20;
}
- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section{
    return (section == self.dataArray.count - 1) ? 10 : 0;
}
@end

File: Freedom/支付宝/我的/SDAssetsTableViewController.swift
//
//  SDAssetsTableViewController.swift
//  Freedom

import UIKit
import BaseFile
import XExtension
class SDAssetsTableViewControllerCell:BaseTableViewCell{
    required init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        textLabel?.textColor = UIColor.darkGray
        textLabel?.font = UIFont.systemFont(ofSize: 15)
        
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func setModel(_ model: NSObject?) {
        let cellModel = model as? SDAssetsTableViewControllerCellModel
        textLabel?.text = cellModel?.title
        imageView?.image = UIImage(named: cellModel?.iconImageName ?? "")
        accessoryType = .disclosureIndicator
    }

}
class SDAssetsTableViewController: AlipayBaseViewController {
    var dataArray = [[SDAssetsTableViewControllerCellModel]]()
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "我的"
        setupModel()
        let header = UIView()
        tableView.tableHeaderView = header
    }
    func setupModel() {
        // section 0 的model
        let model01 = SDAssetsTableViewControllerCellModel(title: "余额宝", iconImageName: "20000032Icon", destinationControllerClass: SDYuEBaoTableViewController.self)
        let model02 = SDAssetsTableViewControllerCellModel(title: "招财宝", iconImageName: "20000059Icon", destinationControllerClass: BaseTableView.self)
        let model03 = SDAssetsTableViewControllerCellModel(title: "娱乐宝", iconImageName: "20000077Icon", destinationControllerClass: BaseTableView.self)
        // section 1 的model
        let model11 = SDAssetsTableViewControllerCellModel(title: "芝麻信用分", iconImageName: "20000118Icon", destinationControllerClass: BaseTableView.self)
        let model12 = SDAssetsTableViewControllerCellModel(title: "随身贷", iconImageName: "20000180Icon", destinationControllerClass: BaseTableView.self)
        let model13 = SDAssetsTableViewControllerCellModel(title: "我的保障", iconImageName: "20000110Icon", destinationControllerClass: BaseTableView.self)

        let model21 = SDAssetsTableViewControllerCellModel(title: "爱心捐赠", iconImageName: "09999978Icon", destinationControllerClass: BaseTableView.self)
        dataArray = [[model01, model02, model03], [model11, model12, model13], [model21]]
    }
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let model = dataArray[indexPath.section][indexPath.row] as? SDAssetsTableViewControllerCellModel
        let vc: UIViewController? = UIViewController()// model?.destinationControllerClass
        vc?.title = model?.title
        if let aVc = vc {
            navigationController?.pushViewController(aVc, animated: true)
        }
    }
    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 20
    }
    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return (section == dataArray.count - 1) ? 10 : 0
    }
}
